Game/Source/GDKTestGyms/Private/BenchmarkNPCCharacter.cpp

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+
+#include "BenchmarkNPCCharacter.h"
+
+
+void ABenchmarkNPCCharacter::OnAuthorityLost()
+{
+	Super::OnAuthorityLost();
+
+	if (Controller)
+	{
+		Controller->Destroy();
```

Comment:
I'm wondering if this is something we should be doing for all AI Characters?

Reply:
Do you mean at some spatial level or project level?

I think you could potentially cache the controller on the actor as a non replicated property. If the NPC comes back to your zone you could reuse the old controller instead of creating a new one. If you loose interest in the NPC though, maybe we should always destroy the controller? I guess you could still cache the NPC entity Id and the controller and match the two back up when it comes back into view...

Reply:
Yeah, I wasn't sure how much of this would be a game-specific implementation, though.